prow/gangway/gangway.go

```diff
@@ -135,13 +147,82 @@ func (gw *Gangway) GetJobExecution(ctx context.Context, gjer *GetJobExecutionReq
 		jobStatus = JobExecutionStatus_JOB_EXECUTION_STATUS_UNSPECIFIED
 
 	}
+	return jobStatus
+}
 
-	jobExec := &JobExecution{
-		Id:        prowJobCR.Name,
-		JobStatus: jobStatus,
+// Translate ProwjobType into JobExecutionType
+func TranslateProwJobType(prowJobType prowcrd.ProwJobType) JobExecutionType {
+	var jobExecutionType JobExecutionType
+
+	switch prowJobType {
+	case prowcrd.PeriodicJob:
+		jobExecutionType = JobExecutionType_PERIODIC
+	case prowcrd.PostsubmitJob:
+		jobExecutionType = JobExecutionType_POSTSUBMIT
+	case prowcrd.PresubmitJob:
+		jobExecutionType = JobExecutionType_PRESUBMIT
+	case prowcrd.BatchJob:
+		jobExecutionType = JobExecutionType_BATCH
+	default:
+		jobExecutionType = JobExecutionType_JOB_EXECUTION_TYPE_UNSPECIFIED
 	}
+	return jobExecutionType
+}
 
-	return jobExec, nil
+func (gw *Gangway) BulkJobStatusChange(ctx context.Context, bjscr *BulkJobStatusChangeRequest) (*JobExecutions, error) {
```

Comment:
Nit: I would name this `request` rather than the acronym

```suggestion
func (gw *Gangway) BulkJobStatusChange(ctx context.Context, request *BulkJobStatusChangeRequest) (*JobExecutions, error) {
```